packages/@ember/-internals/glimmer/tests/integration/application/debug-render-tree-test.ts

```diff
@@ -885,7 +885,7 @@ if (ENV._DEBUG_RENDER_TREE) {
 
         this.addComponent('hello-world', {
           ComponentClass: null,
-          template: 'Hello World',
+          template: '{{@name}}',
```

Comment:
Why do all of these change? It isn't clear why it should have to change from the static string in order to pass the test...

Reply:
So, this is caused by us setting `createInstance` === `false` in TemplateOnly components. The issue is that args are no longer pushed onto the stack unless they are _used_ within the component. So unless we use the arg with `{{@name}}`, it will not appear in the debug render tree.

We could possibly change this behavior in DEBUG only, and just have a different experience in production. I was aiming to unify production and DEBUG behavior here, which is why I made this choice.

Reply:
We chatted a bit about this in DMs. Let's make sure we only modify the tests that are explicitly for template only components (to add the args), that should let us verify that the args are _not_ stripped in debug builds for components with a backing instance.